internal/rbd/rbd_util.go

```diff
@@ -600,36 +600,34 @@ func (rv *rbdVolume) getCloneDepth(ctx context.Context) (uint, error) {
 	vol.Monitors = rv.Monitors
 	vol.RbdImageName = rv.RbdImageName
 	vol.conn = rv.conn.Copy()
-
+	currVol := vol
+	defer currVol.Destroy()
 	for {
-		if vol.RbdImageName == "" {
+		if currVol.RbdImageName == "" {
 			return depth, nil
 		}
-		err := vol.openIoctx()
+		err := currVol.openIoctx()
```

Comment:
Why is opening the iocontext needed? I expect that to be done at the place where it is used.

Reply:
I still do not understand why this is needed. Is it required for calling `getParentVolume()`? If so, it should be done in the function there.

Reply:
Is this needed because the parent and the image pool are different? the openIoctx is per pool?